packages/test/test-end-to-end-tests/src/test/gc/gcSweepDataStores.spec.ts

```diff
@@ -29,48 +31,110 @@ import {
 } from "@fluid-private/test-version-utils";
 import { delay } from "@fluidframework/core-utils";
 import { IContainer, LoaderHeader } from "@fluidframework/container-definitions";
-import { IErrorBase, IFluidHandle } from "@fluidframework/core-interfaces";
-import { getGCDeletedStateFromSummary, getGCStateFromSummary } from "./gcTestSummaryUtils.js";
+import { IErrorBase } from "@fluidframework/core-interfaces";
+// eslint-disable-next-line import/no-internal-modules
+import { RetriableSummaryError } from "@fluidframework/container-runtime/dist/summary/summaryGenerator.js";
+// eslint-disable-next-line import/no-internal-modules
+import { defaultMaxAttemptsForSubmitFailures } from "@fluidframework/container-runtime/dist/summary/runningSummarizer.js";
+// eslint-disable-next-line import/no-internal-modules
+import { ISweepMessage } from "@fluidframework/container-runtime/dist/gc/gcDefinitions.js";
+import {
+	getGCDeletedStateFromSummary,
+	getGCStateFromSummary,
+	manufactureHandle,
+} from "./gcTestSummaryUtils.js";
+
+/**
+ * Validates that the given data store state is correct in the summary.
+ * If expectDelete is true:
+ * - It should be deleted from the data store summary tree.
+ * If expectGCStateHandle is not true:
+ * - It should not be present in the GC state in GC summary tree.
+ * - It should be present in the deleted nodes in GC summary tree.
+ */
+function validateDataStoreStateInSummary(
```

Comment:
I feel the same way :-). We have a plan to go through all the GC tests and update them since the validations have evolved over time. We should make them all use similar validations and share logic as much as possible.